stack/pico_device.c

```diff
@@ -257,14 +252,7 @@ static int devloop_in(struct pico_device *dev, int loop_score)
 
 static int devloop_sendto_dev(struct pico_device *dev, struct pico_frame *f)
 {
-
-    if (dev->eth) {
-        /* Ethernet: pass management of the frame to the pico_ethernet_send() rdv function */
-        return pico_ethernet_send(f);
-    } else {
-        /* non-ethernet: no post-processing needed */
-        return (dev->send(dev, f->start, (int)f->len) <= 0); /* Return 0 upon success, which is dev->send() > 0 */
-    }
+    return (dev->send(dev, f->start, (int)f->len) <= 0);
```

Comment:
No, if the frame gets here, it may have already passed through the link layer. Now this function should be really the single point where a frame leaves the stack (triggered by looping over the devices). Network layers (ipv4.c and ipv6.c) now call ```pico_datalink_send()``` instead of ```pico_sendto_dev(f)```.